// @iobroker/adapter-dev/build/util.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.interceptErrors = exports.getIndentation = exports.die = exports.error = exports.padRight = exports.escapeRegExp = void 0;
const ansi_colors_1 = require("ansi-colors");
function escapeRegExp(value) {
    return value.replace(/[.*+?^${}()|[\]\\]/g, "\\$&"); // $& means the whole matched string
}
exports.escapeRegExp = escapeRegExp;
function padRight(text, totalLength) {
    return text.padEnd(totalLength, " ");
}
exports.padRight = padRight;
function error(message) {
    console.error(ansi_colors_1.bold.red(message));
    console.error();
}
exports.error = error;
function die(message) {
    console.error((0, ansi_colors_1.red)(message));
    process.exit(1);
}
exports.die = die;
function getIndentation(text) {
    const lines = text.split(/\r?\n/);
    const fileStartLine = lines.findIndex((line) => line.startsWith("{"));
    if (fileStartLine !== -1 && lines.length > fileStartLine) {
        const matches = lines[fileStartLine + 1].match(/^[ ]+/);
        console.log(matches);
        if (matches && matches.length >= 1) {
            return matches[0].length;
        }
    }
    return 4;
}
exports.getIndentation = getIndentation;
function interceptErrors(func) {
    return async () => {
        try {
            await func();
        }
        catch (error) {
            die(error.stack || error);
        }
    };
}
exports.interceptErrors = interceptErrors;
